lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -142,4 +142,34 @@ module.exports = function($q, $uibModalInstance, DrivesModel, SelectionStateMode
     });
   };
 
+  this.getButtonStatus = () => {
+    const isDisabled = !this.state.hasDrive() || this.state.getDrive().protected;
+
+    if (isDisabled) {
+      return 'disabled';
+    }
+
+    const isDangerous = this.state.hasDrive() && this.state.isSystemDrive(this.state.getDrive());
+
+    if (isDangerous) {
+      return 'danger';
+    }
+
+    return 'primary';
+  };
+
+  this.getFooterMessage = () => {
+    if (this.getButtonStatus() === 'danger'
+      && this.state.isSystemDrive(this.state.getDrive())) {
+      return ManifestBindService.get('displayName') + ' has identified this as a system drive!';
+
+    }
+
+    return undefined;
+  };
+
+  this.hasFooterMessage = () => {
```

Comment:
This function seems to be completely unused, in which case we can remove it and the tests as well.